main.go

```diff
@@ -76,7 +61,8 @@ func main() {
 	log.Info("server stopped", "port", port)
 }
 
-func logMiddleware(next http.HandlerFunc) http.HandlerFunc {
+// log is a middleware that logs all incoming requests
+func logMdl(next http.HandlerFunc) http.HandlerFunc {
```

Comment:
"middleware," allowing me to wrap all routes with a function that logs details, so I only need to write it once. can be used for authorizations, too.

The "decorator" (e.g. @login_required) in python is the same idea. The middleware goes first, then the function can continue after actions or checks pass.